### ISSUE TITLE
(sdcc) Generates dubious code for ez80_z80

### ISSUE THREAD

**Issue by suborb** [z88dk/z88dk#2303]
It looks md5sum generates bad code when compiled with sdcc. It appears to assume that the following:

```
ld (ix+n),0x0000
```

is equivalent to:

```
xor a
ld (ix+n),a
ld (ix+n+1),a
```

Trying to track down whether it's a rules problem on our side...

It's this one from sdcc_peeph.2:

```
replace restart {
        ld      %1, %2
        ld      %3 (%4), %1
} by {
        ; peephole 12 loaded %2 into %3 (%4) directly instead of going through %1.
        ld      %3 (%4), %2
} if canAssign(%3 %4 %2), notVolatile(%1), notUsed(%1)
```

I've checked against the latest sdcc and there's been a few fixes in canAssign since our version and this issue no longer arises.

**Comment by suborb** [comment]
Linking into #2186 to close with the upgrade.

**Comment by feilipu** [comment]
Closed by #2190

**Comment by suborb** [comment]
I obviously compiled sdcc without our rules....this is still happening
